src/containers/Tenant/Query/QueryResult/components/PlanToSvgButton/PlanToSvgButton.tsx

```diff
@@ -24,20 +25,49 @@ export function PlanToSvgButton({plan, database}: PlanToSvgButtonProps) {
     const [blobUrl, setBlobUrl] = React.useState<string | null>(null);
     const [getPlanToSvg, {isLoading}] = planToSvgApi.useLazyPlanToSvgQueryQuery();
 
-    const handleClick = React.useCallback(() => {
-        getPlanToSvg({plan, database})
+    const handleGetSvg = React.useCallback(() => {
+        if (blobUrl) {
+            return Promise.resolve(blobUrl);
+        }
+
+        return getPlanToSvg({plan, database})
             .unwrap()
             .then((result) => {
                 const blob = new Blob([result], {type: 'image/svg+xml'});
                 const url = URL.createObjectURL(blob);
                 setBlobUrl(url);
                 setError(null);
-                window.open(url, '_blank');
+                return url;
             })
             .catch((err) => {
-                setError(JSON.stringify(err));
+                const errorMessage = err.data?.message || err.message || JSON.stringify(err);
```

Comment:
Don't we want to move such a logic into a helper (we have similar logic in `ResponseError` and `parseQueryError`)?

Reply:
moved to utils/errors